# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.cpp
# Commit: b620533c6d63
# Full Hash: b620533c6d633ef50c5e5cb3ecd0f398206eb271
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-06-17 21:52:06
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Refactor wasm::Frame to reduce number of gnarly casts. r=lth,wingo
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D79072
# ==============================================================================

diff -r f7ec4a920540 -r b620533c6d63 js/src/jit/MacroAssembler.cpp
--- a/js/src/jit/MacroAssembler.cpp	Wed Jun 17 11:41:13 2020 +0000
+++ b/js/src/jit/MacroAssembler.cpp	Wed Jun 17 12:04:54 2020 +0000
@@ -3700,9 +3700,8 @@
 }
 
 void MacroAssembler::loadWasmTlsRegFromFrame(Register dest) {
-  loadPtr(
-      Address(getStackPointer(), framePushed() + offsetof(wasm::Frame, tls)),
-      dest);
+  loadPtr(Address(getStackPointer(), framePushed() + wasm::Frame::tlsOffset()),
+          dest);
 }
 
 void MacroAssembler::BranchGCPtr::emit(MacroAssembler& masm) {